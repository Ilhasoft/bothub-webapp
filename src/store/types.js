export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
