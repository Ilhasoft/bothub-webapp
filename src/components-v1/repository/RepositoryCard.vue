<template>
  <div class="card">
    <div class="card-content">
      <repository-info
        :slug="slug"
        :name="name"
        :owner__nickname="owner__nickname"
        :available_languages="available_languages"
        :description="description"
        :categories_list="categories_list"
        :votes_sum="votes_sum" />
    </div>
  </div>
</template>

<script>
import RepositoryInfo from '@/components-v1/repository/RepositoryInfo';

const components = {
  RepositoryInfo,
};

export default {
  name: 'RepositoryCard',
  components,
  props: {
    slug: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    owner__nickname: {
      type: String,
      required: true,
    },
    available_languages: {
      type: Array,
      default: () => ([]),
    },
    description: {
      type: String,
      default: '',
    },
    categories_list: {
      type: Array,
      default: () => ([]),
    },
    votes_sum: {
      type: Number,
      default: 0,
    },
  },
};
</script>

<style lang="scss" scoped>
.card {
  margin: 1rem 0;
}
</style>
