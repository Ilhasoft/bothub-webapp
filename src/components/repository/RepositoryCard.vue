<template>
  <div
    :class="[ single ? 'repository-card__single' : 'repository-card']">
    <div
      id="tour-create_intelligence-step-0"
      class="repository-card__align-items">
      <div
        class="repository-card__big-badge-wrapper">
        <component
          :is="clickable ? 'router-link' : 'span'"
          :to="repositoryDetailsRouterParams">
          <div class="repository-card__big-badge">
            <custom-icon
              :value="repositoryIcon"
              :class="{'repository-card__big-badge__icon': true, clickable }"
              size="extra-large" />
          </div>
        </component>
      </div>
      <div class="repository-card__details">
        <div class="repository-card__title">
          <span class="repository-card__title__bagde">
            <component
              :is="clickable ? 'router-link' : 'span'"
              :to="repositoryDetailsRouterParams">
              <custom-icon
                :fillMobile="true"
                :value="repositoryIcon"
                :class="{'repository-card__title__bagde__icon': true, clickable}" /></component>
          </span>
          <component
            :is="clickable ? 'router-link' : 'span'"
            :to="repositoryDetailsRouterParams">
          <span class="repository-card__title__repository">{{ $attrs.name }}</span></component>
        </div>

        <div class="repository-card__info-item">
          <span>{{ $t('webapp.layout.created_by') }}</span>
          <strong class="repository-card__info-item__created-by">
            {{ getProfile($attrs.owner__nickname).name || $attrs.owner__nickname }}
          </strong>
        </div>
        <div class="repository-card__flags">
          <span
            v-for="language in $attrs.available_languages"
            :key="language"
            :class="{
              'repository-card__flags__flag': true,
              'repository-card__flags__flag--main': language == $attrs.language,
          }">
            <language-badge
              :language="language"
              :main="language == $attrs.language" />
          </span>
        </div>
        <div class="repository-card__categories">
          <component
            v-for="category in repositoryCategoryRouterParams"
            :is="clickable ? 'router-link' : 'span'"
            :key="category.id"
            :to="{
              path: '/home',
              query: { category: category.id },
          }">
            <b-tag
              color="grey-light"
              rounded
              size="is-small"
              class="repository-card__categories__category">{{ category.name }}</b-tag>
          </component>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import LanguageBadge from '@/components/shared/LanguageBadge';
import CustomIcon from '@/components/shared/CustomIcon';

export default {
  name: 'RepositoryCard',
  components: {
    LanguageBadge,
    CustomIcon,
  },
  props: {
    single: {
      type: Boolean,
      default: null,
    },
    clickable: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    ...mapGetters([
      'getProfile',
    ]),
    repositoryDetailsRouterParams() {
      return {
        name: 'repository-summary',
        params: {
          ownerNickname: this.$attrs.owner__nickname,
          slug: this.$attrs.slug,
        },
      };
    },
    repositoryCategoryRouterParams() {
      if (typeof this.$attrs.categories[0] === 'object') {
        return this.$attrs.categories;
      }
      if (typeof this.$attrs.categories_list[0] === 'object') {
        return this.$attrs.categories_list;
      }
      return [];
    },
    repositoryIcon() {
      return ((this.$attrs.categories[0] && this.$attrs.categories[0].icon)
      || (this.$attrs.categories_list[0] && this.$attrs.categories_list[0].icon))
      || 'botinho';
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@/assets/scss/variables.scss';
@import '~@/assets/scss/colors.scss';


.repository-card {
  $repository-card-margin: .5rem;

  margin: $repository-card-margin 0 $repository-card-margin 2*$repository-card-margin;
  padding: 1rem 0;
  width: calc(100% * 1/3 - 1rem);
  word-break: break-all;
  background-color: white;
  border: 1px solid transparent;
  border-radius: 0.5rem;
  box-shadow:0rem 0rem 0.5rem 0.2rem rgba(0, 0, 0, 0.1);

  @media screen and (max-width: $mobile-width) {
    width: calc(100% * 1/2 - 1rem);
  }

  @media screen and (max-width: $small-mobile-width) {
    width: calc(100% - 1rem);
    margin: $repository-card-margin auto;
  }

  &__single {
    width: 100%;
    height: 100%;
    padding: 1rem 0;
    word-break: break-all;
    background-color: white;
    border: 1px solid transparent;
    border-radius: 0.5rem;
    box-shadow:0rem 0rem 0.5rem 0.2rem rgba(0, 0, 0, 0.1);
  }

  &__align-items {
    align-items: center;
    display: flex;
    padding: 0.25rem;
    flex-direction: column;
  }

  &__details {
    text-align: center;
    padding: 0.5rem;
  }

  &__big-badge-wrapper {
    display: block;
    margin: 0.5rem;
    flex-grow: 0;
    flex-shrink: 1;
    flex-basis: auto;

    @media screen and (max-width: 800px) {
      display: none;
    }
  }

  &__big-badge {
    $size: 6rem;

    position: relative;
    display: block;
    width: $size;
    height: $size;
    border-radius: 50%;
    background-color: $color-primary-dark;
    overflow: hidden;

    &__icon {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }

  &__title {
    font-size: 1.25rem;
    font-weight: $font-weight-bolder;
    margin-bottom: .5rem;

    &__bagde {
      display: none;
      vertical-align: middle;
      margin-right: .5rem;

      &__icon {
        font-size: 1.5em;
      }

      @media screen and (max-width: $mobile-width) {
        display: inline-block;
      }
    }

    &__repository{
      color: $color-fake-black;
      font-family: $font-family;
      font-weight: $font-weight-bolder;
    }
  }

  &__info-item {
    color: $color-grey-dark;

    &__created-by{
      color: $color-primary;
      font-weight: $font-weight-normal;
    }
  }

  &__flags {
    display: flex;
    flex-wrap: wrap;
    margin: .5rem -.25rem;
    justify-content: center;
    align-items: center;

    &__flag {
      margin: .25rem;
      order: 0;

      &--main {
        order: -1;
      }
    }
  }

  &__categories {
    margin: .5rem -.25rem;

    &__category {
      margin: .25rem;
      padding: 0 1rem;
    }
  }
}
</style>
