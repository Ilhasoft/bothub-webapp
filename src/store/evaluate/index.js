import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
