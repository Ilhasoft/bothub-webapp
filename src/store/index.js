import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
