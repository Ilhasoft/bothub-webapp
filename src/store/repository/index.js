import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
