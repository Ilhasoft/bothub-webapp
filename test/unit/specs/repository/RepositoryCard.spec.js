/* eslint-disable import/first */
jest.mock('@/api/request');

import Vuex from 'vuex';
import Buefy from 'buefy';
import { shallow, createLocalVue } from '@vue/test-utils';
import RepositoryCard from '@/components/repository/RepositoryCard';

import store from '@/store';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Buefy);


describe('RepositoryCard.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(RepositoryCard, {
      localVue,
      store,
      propsData: {
        repository: {
          uuid: 'dbe1539c-88ba-43e0-bed5-54729218cec9',
          owner: 1,
          owner__nickname: 'fake',
          name: 'Repository',
          slug: 'repository',
          language: 'en',
          available_languages: ['en', 'pt'],
          categories: [1, 2],
          categories_list: [
            { id: 1, name: 'Business' },
            { id: 2, name: 'Technology' },
          ],
          description: '',
          is_private: false,
          created_at: '2018-04-20T17:05:42.035514Z',
        },
      },
    });
  });

  test('mount', () => {
    expect(wrapper.vm).toBeDefined();
  });
});
