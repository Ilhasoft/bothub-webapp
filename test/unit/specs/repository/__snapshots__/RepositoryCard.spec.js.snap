// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard.vue renders correctly 1`] = `
<div class="card">
  <div class="card-content">
    <repositoryinfo-stub></repositoryinfo-stub>
  </div>
</div>
`;
